### PR TITLE
Big Bang 9: Restructure activity animation graph

### LINKED ISSUE #56: Improve colormap/scale options
Add unit, etc. (feedback from the Hackathon)

### PR DESCRIPTION
**Pull request by babsey**
This PR restructures the activity animation graph, which allows configuring multiple layers
 indicating various activities from recorders.

image

closes #56

### CODE REVIEW

File: src/core/activity/activityAnimation/activityAnimationLayerModel.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      mesh.scale.set(scale, scale, scale);
+    });
+    this._layer.state.reset = true;
+  }
+
+  /**
+   * Update graph objects.
+   */
+  updateObjects(frame: any, trailIdx: number = null): void {
+    // console.log('Update objects');
```
Comment (jens-br) [inline src/core/activity/activityAnimation/activityAnimationLayerModel.ts:87]: If it is unlikely that it is used again, I would opt for removing it.
```suggestion
```

File: src/core/activity/activityGraph.ts [hunk 2]
```diff
@@ -61,7 +61,11 @@ export class ActivityGraph {
    * Initialize activity animation graph (Three).
    */
   initActivityAnimationGraph(): void {
-    this._activityAnimationGraph = new ActivityAnimationGraph(this._project);
+    if (this._activityAnimationGraph == undefined) {
```
Comment (jens-br) [inline src/core/activity/activityGraph.ts:64]:
```suggestion
    if (this._activityAnimationGraph == null) {
```

**Review by jens-br — CHANGES_REQUESTED** [review]
Good idea, some minor adjustments suggested.

File: src/core/activity/activityAnimation/activityAnimationGraph.ts [hunk 3]
```diff
@@ -254,165 +70,152 @@ export class ActivityAnimationGraph {
...
-      this.updateLayers(activity);
-      this.updateFrames(activity);
+      let layer = this._layers[activity.idx];
+      if (layer == null) {
+        layer = new ActivityAnimationLayer(this, activity);
+        this._layers.push(layer);
+      } else if (layer.activity !== activity) {
+        layer = new ActivityAnimationLayer(this, activity);
+        this._layers[activity.idx] = layer;
+      }
```
[the remark this answers is not in the document]
Comment (babsey) [reply src/core/activity/activityAnimation/activityAnimationGraph.ts]: The code suggestion is strange, thus I made own commit with very same code with only difference `const layer = ...`

**Review by jens-br — APPROVED** [review]
LGTM :+1:

**Pull request merged**